.github/actions/override-workflow-variables/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Override Workflow Variables'
+description: |-
+  Use this action to load/override variables into GITHUB_ENV.
+
+inputs:
+  working_directory:
+    description: 'The working directory to look for the env file.'
+    required: false
+    default: '.'
+  env_file:
+    description: 'The env file containing variables to override with.'
+    required: false
+    default: '.env'
+  require_env:
+    description: 'Require the use of an env file for overriding variables.'
+    required: false
+    default: 'false'
+  enforced_directories:
+    description: 'A CSV string of directories to require the use of an env file for overriding variables.'
```

Comment:
I also had a hard time understanding this, maybe considering something like `only used when require_env=true; rather than looking for an env file in every immediate subdirectory, just require it in this set of directories`.